src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -2231,107 +2240,111 @@ abstract class ManagedStrategy(
         return threadId in (0 ..< nThreads)
     }
 
-    /**
-     * Logs thread events such as thread switches and passed code locations.
-     */
-    private inner class TraceCollector {
-        private val _trace = mutableListOf<TracePoint>()
-        val trace: List<TracePoint> = _trace
-        private var spinCycleStartAdded = false
-
-        private val spinCycleMethodCallsStackTraces: MutableList<List<CallStackTraceElement>> = mutableListOf()
-
-        fun newSwitch(iThread: Int, reason: SwitchReason, beforeMethodCallSwitch: Boolean = false) {
-            if (reason == SwitchReason.ActiveLock) {
-                afterSpinCycleTraceCollected(
-                    trace = _trace,
-                    callStackTrace = callStackTrace[iThread]!!,
-                    spinCycleMethodCallsStackTraces = spinCycleMethodCallsStackTraces,
-                    iThread = iThread,
-                    currentActorId = currentActorId[iThread]!!,
-                    beforeMethodCallSwitch = beforeMethodCallSwitch
-                )
-            }
-            val callStackTrace = when (reason) {
-                SwitchReason.Suspended -> suspendedFunctionsStack[iThread]!!.reversed()
-                else -> callStackTrace[iThread]!!
+    // == TRACE COLLECTOR EXTENSION METHODS ==
+    private fun TraceCollector.passCodeLocation(tracePoint: TracePoint?) {
+        if (tracePoint !is SectionDelimiterTracePoint && tracePoint?.isActorMethodCallTracePoint() == false) {
+            checkActiveLockDetected()
+        }
+
+        if (tracePoint != null) {
```

Comment:
What about just adding `if (tracePoint == null return` at the beginning of the method body?